### PR TITLE
feat(utility): add File class to represent Graviti files

### PR DESCRIPTION
**Pull request by Lee-000**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2b4rty0 实现 File 类，支持通过 url 读取远端文件](https://app.clickup.com/t/2b4rty0)

File: graviti/utility/file.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    """
+
+    def __init__(self, url: str, checksum: str) -> None:
+        super().__init__("")
+        self.url = _URL(url)
+        self._checksum = checksum
+
+    def _repr_head(self) -> str:
+        return f'{self.__class__.__name__}("{self._checksum}")'
```
Comment (linjiX) [inline graviti/utility/file.py:30]: will the full checksum too long？will it be better if we just display the first 7~8 letters of the checksum?

File: graviti/utility/file.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+class File(RemoteFileMixin):
+    """This class represents the file on Graviti platform.
+
+    Arguments:
+        url: The url string of the file.
+        checksum: The checksum of the file.
+
+    """
+
```
Comment (linjiX) [inline graviti/utility/file.py:23]: shall we use `__slots__` to save memory?

File: graviti/utility/file.py (hunk 1 above)
Comment (linjiX) [reply graviti/utility/file.py:30]: @AChenQ

Comment (AChenQ) [reply graviti/utility/file.py:30]: We can just display the first 8 letters of the checksum now.

File: graviti/utility/file.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    __slots__ = ("url", "_checksum")
+
+    _show_checksum_length = 8
```
Comment (linjiX) [inline graviti/utility/file.py:26]: upper case for constant?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/utility/file.py (hunk 3 above)
Comment (linjiX) [reply graviti/utility/file.py:26]: can we reuse tensorbay `short` function?

**Review by linjiX — CHANGES_REQUESTED** [review]

**Review by linjiX — APPROVED** [review]

**Review by AChenQ — APPROVED** [review]

**Pull request merged**